tester/runner.go

```diff
@@ -236,6 +249,19 @@ func (r *Runner) runTest(ctx context.Context, mod *ast.Module, rule *ast.Rule) (
 		}
 	} else if len(rs) == 0 {
 		tr.Fail = true
+		if bufFailureLineTracer != nil {
+			events := *bufFailureLineTracer
+			eventsLen := len(events)
+			for i, opFail := eventsLen-1, 0; i >= 0; i-- {
+				if strings.Compare(string(events[i].Op), "Fail") == 0 {
+					opFail++
+				}
+				if opFail == SecondToLast {
```

Comment:
After testing this manually (by causing a failure in some of the open-policy-agent/librarytests) I realized that we'll need to handle the case where there is only one Fail event in the trace. This can occur if the rule indexing optimization kicks in when the test rule is evaluated.

For example:

```
package x

test_foo {
  input.x = 1  # indexer understands this
}
```

When OPA runs the query `data.x.test_foo = result`, the indexer will return 0 rules (since `input` is not defined) and the body will not be evaluated at all. As a result the trace will only have a single FAIL event. It'll be something like:

```
Enter data.x.test_foo = result      # entering query
  Eval data.x.test_foo = result     # evaluating expression
  Index data.x.test_foo (0 results) # using rule index
  Fail data.x.test_foo = result     # single Fail event since the expression is undefined (because no rules matched)
```

We'll need to handle that here.

Reply:
Besides the message "(matched 0 rules)" is there anything else that tells us the test was skipped? I looked at the event trace for both cases and they (seem) the same.

Reply:
Not really. We'll just need to handle the single Fail case described above.

Reply:
Done